payments/utils.py

```diff
@@ -0,0 +1,134 @@
+
+def round_price(price):
+    return price.quantize(Decimal("0.01"), rounding=ROUND_HALF_UP)
+
+
+def rounded(func):
+    """
+    Decorator for rounding function result
+    """
+
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        value = func(*args, **kwargs)
+        return round_price(value)
+
+    return wrapped
+
+
+def currency_format(value):
+    return f"{round_price(value)}€"
+
+
+def currency(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        value = func(*args, **kwargs)
+        return currency_format(value)
+
+    return wrapped
+
+
+def percentage_format(value):
+    return f"{round_price(value)}%"
+
+
+def percentage(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        value = func(*args, **kwargs)
+        return percentage_format(value)
+
+    return wrapped
+
+
+def convert_aftertax_to_pretax(aftertax_price, tax_percentage):
+    return aftertax_price / (1 + tax_percentage / 100)
+
+
+def calculate_order_due_date(delta=14):
+    return date.today() + timedelta(days=delta)
+
+
+def _calculate_price_for_partial_month(
+    month_date: date, base_price: Decimal
+) -> Decimal:
+    _, days_in_month = calendar.monthrange(month_date.year, month_date.month)
+    price_per_day = base_price / days_in_month
+    return (days_in_month - month_date.day) * price_per_day
+
+
+@rounded
+def calculate_product_partial_month_price(
+    base_price: Decimal, start_date: date, end_date: date
+) -> Decimal:
+    price = Decimal(0)
+
+    # If the dates are on the same month/year
+    if start_date.month == end_date.month and start_date.year == end_date.year:
+        price = _calculate_price_for_partial_month(start_date, base_price)
+    else:
+        # Calculate price for the start month
+        price += _calculate_price_for_partial_month(start_date, base_price)
+
+        # Calculate price for the months in between
+        months_between = list(
+            rrule(
+                MONTHLY,
+                dtstart=start_date + relativedelta(day=1, months=1),
+                until=end_date - relativedelta(day=1, months=1),
+            )
+        )
+        # Calculate the added price for all the complete months in between
+        price += len(months_between) * base_price
+
+        # Calculate price for the end month
+        price += _calculate_price_for_partial_month(end_date, base_price)
+
+    return price
+
+
+@rounded
+def calculate_product_partial_season_price(
+    base_price: Decimal, start_date: date, end_date: date, summer_season: bool = True
+) -> Decimal:
+    # If it's the "normal" (summer) season, calculate with the normal dates
+    season_days = calculate_berth_lease_end_date() - calculate_berth_lease_start_date()
+    # If it's for the opposite season ("winter season"), calculate the inverse
+    if not summer_season:
+        # Calculate the amount of days in the year, in case it's a leap year
+        day = today()
+        days_on_year = (day + relativedelta(years=1)) - day
```

Comment:
`days_IN_year`? 🤔

Reply:
🇬🇧

Reply:
[google not sure](https://www.google.com/search?sxsrf=ALeKk02qZSkRC2VFP43qfE5qwJqQ3PW6-w%3A1591879522792&ei=YifiXqbhL6qqrgSY1JPACQ&q=%22days+on+year%22&oq=%22days+on+year%22&gs_lcp=CgZwc3ktYWIQAzIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeOgQIABBHOgQIABBDOgYIABAHEB46BAgAEApQl94PWMvxD2CR9A9oAHABeACAAXWIAdcBkgEDMS4xmAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwimmuyr5fnpAhUqlYsKHRjqBJgQ4dUDCAw&uact=5)

Reply:
🤷 days`days_IN_year` works fine.